frontend/src/components/explorer/mapViewer/DataOverlay.vue

```diff
@@ -213,6 +213,9 @@ export default {
       this.showFileLoader = true;
     });
   },
+  updated() {
+    setTimeout(() => document.getElementById('searchInput').focus());
+  },
```

Comment:
Good catch on this interaction too. The `RNAexpression` component, via the `DataOverlay` component, currently uses `EventBus` to broadcast changes with RNA level application, using the line:

```javascript
EventBus.$emit(this.showing2D ? 'apply2DHPARNAlevels' : 'apply3DHPARNAlevels', {});
```

You can see how this is handled in `Svgmap`.

```javascript
  created() {
    EventBus.$off('apply2DHPARNAlevels');
    EventBus.$on('apply2DHPARNAlevels', (levels) => {
      this.applyHPARNAlevelsOnMap(levels);
    });
```

Using a similar pattern, I think it would be better to handle the `EventBus` message inside the `MapSearch` component so that it can focus on the `searchInput` by itself.

Two things I want to mention about this suggestion:

1. It seems like quite a bit more work to achieve the same goal but this keeps components more modular. The `DataOverlay` should not directly manipulate other components.
2. The `EventBus` is something we have been gradually trying to remove and refactor by using Vuex because we believe that in general Vuex provides a better and more explicit (less guesswork) design pattern. Having said that, there is still currently a lot of `EventBus` usage especially for dealing with RNA levels. So I think for now it is still recommendable to use `EventBus` for this case, until we plan a more thorough refactor work.

Reply:
Hi @shan-chalmers, thanks for the explanation and I understand slightly better with the relationships of these components and hooks after the debugging process. For this part of code, I have tried to replace the code `setTimeout(() => document.getElementById('searchInput').focus());` by reusing the function `focusOnInputSearch()` from `MapSearch`. However, it just does not work. Even I define the same function inside `DataOverlay.vue` but still not working. Feels a bit weird.

Reply:
Sorry, the `EventBus` stuff definitely is outside of our usual work with Vue, I should have been a bit more detailed in the explanation.

I think it would make sense to add the following to the `MapSearch` component:
```javascript
created() {
    EventBus.$off('apply2DHPARNAlevels');
    EventBus.$on('apply2DHPARNAlevels', () => {
      this. focusOnInputSearch();
    });
}
```

Is that what you tried? I haven't tried this myself so not 100% sure that it works but I think it should do the job.